ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -79,8 +80,13 @@ S3Connection::ListObjects(const std::string &prefix, std::vector<std::string> &o
     listSingle ? request.SetMaxKeys(1) : request.SetMaxKeys(batchSize);
 
     Aws::S3Crt::Model::ListObjectsV2Outcome outcomes = _s3CrtClient.ListObjectsV2(request);
-    if (!outcomes.IsSuccess())
-        return (1);
+    Aws::Http::HttpResponseCode resCode = outcomes.GetError().GetResponseCode();
+    if (!outcomes.IsSuccess()) {
+        if (toErrno.find(resCode) != toErrno.end())
+            return (toErrno.at(resCode));
+        else
+            return (1);
```

Comment:
I'm not thrilled about returning `1` by default as it corresponds to a system error. Maybe return `-1`? @sulabhM what do you think?

Reply:
Ya, return of 1 will be "EPERM 1 Operation not permitted", lets return -1

Reply:
Yeah, that makes sense, does WiredTiger do anything in particular for handling a return value of -1?